lib/Jit/utility.cpp

```diff
@@ -1,59 +1,187 @@
-//===----------------- lib/Jit/utility.cpp ----------------------*- C++ -*-===//
-//
-// LLILC
-//
-// Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license.
-// See LICENSE file in the project root for full license information.
-//
-//===----------------------------------------------------------------------===//
-///
-/// \file
-/// \brief Definitions of utility classes.
-///
-//===----------------------------------------------------------------------===//
+// Utility code
+
+#include <cstdlib>
 
 #include "global.h"
 #include "jitpch.h"
+
+#include "cor.h" // CorSigUncompressData
 #include "utility.h"
 #include "llvm/Support/ConvertUTF.h"
 
-using namespace llvm;
+using std::string;
+using std::unique_ptr;
+
+int MethodID::parseArgs(const string &S, size_t &I) {
+  int NArgs = 0;
+  size_t SLen = S.length();
+
+  ++I;                          // skip (
+  string num = scan(S, I);      // eg: 3)
+  if (I >= SLen || S[I] != ')') // illegal
+    return MethodIDState::BAD;
+
+  NArgs = atoi(num.c_str());
+  ++I; // skip )
+  return NArgs;
+}
+
+void MethodID::skip(const string &S, size_t &I) {
+  NumArgs = MethodIDState::BAD;
+  size_t SLen = S.length();
+
+  if (SLen == 0) { // empty string: nothing to do
+    NumArgs = MethodIDState::EMPTY;
+    return;
+  }
+
+  for (; I < SLen && S[I] == ' '; ++I) // skip leading whitespace
+    ;
+  if (I >= SLen) { // S was all-blank
+    NumArgs = MethodIDState::EMPTY;
+  }
+}
+
+MethodID::MethodID(const string &S, size_t &I)
+    : ClassName(nullptr), MethodName(nullptr),
+      NumArgs(MethodIDState::BAD) { // assume a bad parse
+
+  size_t SLen = S.length();
```

Comment:
Parsing the string should be a different function; doesn't seem to fit in a constructor.